### PR TITLE
Enable PWM for stm32l562e_dk

### PR DESCRIPTION
**Pull request by sarthsmart**
These commits enable PWM in device tree, drivers & tests for stm32l562e_dk platform.

Verified PWM signals on logic analyzer by running `tests/drivers/pwm/pwm_api` sample.

### CODE REVIEW

**Review by MaureenHelm — APPROVED** [review]

**Review by erwango — APPROVED** [review]

File: dts/arm/st/l5/stm32l5.dtsi [hunk 1]
```diff
@@ -246,6 +246,168 @@
 			status = "disabled";
 			label = "RNG";
 		};
+
+		timers1: timers@50012c00 {
```
Comment (erwango) [inline dts/arm/st/l5/stm32l5.dtsi:250]: You're using secure address space (`_S`)
This is wrong as Zephyr application isn't secure.
You should use non secure `_NS` address ranges

Comment (ABOSTM) [inline dts/arm/st/l5/stm32l5.dtsi:250]: 0x50012c00 is the address for secure world. but Zephyr is running in non secure.
```suggestion
		timers1: timers@40012c00 {
```
The same for all timers

**Review by erwango — CHANGES_REQUESTED** [review]

File: dts/arm/st/l5/stm32l5.dtsi [hunk 2]
```diff
@@ -246,6 +246,168 @@
...
+			compatible = "st,stm32-timers";
+			reg = <0x50012c00 0x400>;
```
Comment (ABOSTM) [inline dts/arm/st/l5/stm32l5.dtsi:252]: Use non secure address

**Review by ABOSTM — CHANGES_REQUESTED** [review]

File: dts/arm/st/l5/stm32l5.dtsi (hunk 1 above)
Comment (sarthsmart) [reply dts/arm/st/l5/stm32l5.dtsi]: Resolved.

Comment (sarthsmart) [reply dts/arm/st/l5/stm32l5.dtsi]: Resolved.

**Review by ABOSTM — APPROVED** [review]
LGTM

File: tests/drivers/pwm/pwm_api/boards/stm32l562e_dk.overlay [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (c) 2021 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&timers2 {
+	pwm {
+		st,prescaler = <10000>;
+	};
```
Comment (FRASTM) [inline tests/drivers/pwm/pwm_api/boards/stm32l562e_dk.overlay:10]: according to the 'dts/arm/st: Default "st,prescaler" pwm property to 0' #36133
should the prescaler now be `<0>`

Comment (sarthsmart) [reply tests/drivers/pwm/pwm_api/boards/stm32l562e_dk.overlay]: > according to the 'dts/arm/st: Default "st,prescaler" pwm property to 0' #36133
> should the prescaler now be `<0>` instead ?

According to [https://github.com/zephyrproject-rtos/zephyr/pull/36133](https://github.com/zephyrproject-rtos/zephyr/pull/36133), added prescaler value <10000> to `stm32l562e_dk.dts` after removing overlay file changes,
the PWM duty cycle forms  `0%... 25%... 50%... 75%...` for pwm api tests on USB logic analyzer.

**Review by erwango — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Comment by galak** [comment]
Can you rebase your tree on a recent `main` branch of zephyr.  I think that will help get the CI issues to pass.

**Comment by sarthsmart** [comment]
Open a new PR [https://github.com/zephyrproject-rtos/zephyr/pull/36384](https://github.com/zephyrproject-rtos/zephyr/pull/36384)

**Pull request closed without merging**